### PR TITLE
Adds rough W3C spec version.

### PR DESCRIPTION
**Pull request by staats-google**
This probably doesn't work perfectly, but it's good enough to start with. The basic_sync test now passes on Firefox.

Will follow up by: 
  - hammering each piece into place with tests
  - updating the error handling code
  - adding an Actions API to handle keyboard/mouse events (which are totally different).

### CODE REVIEW

File: lib/src/sync/w3c_spec/web_element.dart [hunk 1]
```diff
@@ -0,0 +1,156 @@
...
+  @override
+  bool operator ==(other) =>
+      other is WebElement && other.driver == this.driver && other.id == this.id;
+
+  @override
+  String toString() {
+    final out = new StringBuffer()..write(context);
+    if (locator is By) {
+      if (index == null) {
+        out..write('.findElement(');
```
Comment (manuelF) [inline lib/src/sync/w3c_spec/web_element.dart]: prefer . when you are not actually cascading.

File: lib/src/sync/window.dart [hunk 2]
```diff
@@ -28,11 +28,11 @@ abstract class Windows {
 /// Upon use, the window will automatically be set as active.
 abstract class Window {
   /// The size of the window.
-  @Deprecated('JSON wire legacy support, emulated for newer browsers')
+  @Deprecated('JSON wire legacy support, unsupported for newer browsers')
```
Comment (manuelF) [inline lib/src/sync/window.dart]: What does unsupported here means? if it truly doesn't work, why cant we just remove this?

File: lib/sync_core.dart [hunk 3]
```diff
@@ -53,24 +61,52 @@ WebDriver createDriver({Uri uri, Map<String, dynamic> desired}) {
...
+      final processor =
+          new SyncHttpCommandProcessor(processor: processW3cResponse);
+      final response = processor.get(uri.resolve('session/$sessionId'))
+          as Map<String, dynamic>;
+      return new w3c.W3cWebDriver(
+          processor, uri, sessionId, new UnmodifiableMapView(response));
+    case WebDriverSpec.Auto:
+      throw 'Not yet supported!';
+  }
+  throw 'Inconcievable!'; // Really really.
```
Comment (manuelF) [inline lib/sync_core.dart]: Better error message please. :)

File: lib/sync_io.dart [hunk 4]
```diff
@@ -14,8 +14,7 @@
 
 library webdriver.sync_io;
 
-import 'dart:io' show ContentType, HttpHeaders;
-
+import 'package:webdriver/sync_core.dart' as core;
```
Comment (manuelF) [inline lib/sync_io.dart]: remove this import and extend the show on the other one.

File: lib/src/sync/window.dart (hunk 2 above)
Comment (staats-google) [inline lib/src/sync/window.dart]: The JSON wire spec specifies this in terms of size and location, but W3C specifies it in terms of rect. So location/size are what people use now, and rect is what they should use in the future.

To use the same interface for all WebDriver tests (i.e. irrespective of browser) we need the API there at least, even if it's not hooked up.  I could emulate size and location for W3C, but as these are poorly specified anyway I want people to move to rect. (Also emulation would be a 50-100ms slower due to extra calls.) 

In practice, I suspect this is non issue -- I don't think anyone uses this much. But in the event that people do, they have a migration path and we can drop this later.

File: lib/sync_core.dart (hunk 3 above)
Comment (staats-google) [inline lib/sync_core.dart]: Ah you're no fun.

File: lib/sync_io.dart (hunk 4 above)
Comment (staats-google) [inline lib/sync_io.dart]: Done.

**Comment by staats-google** [comment]
Couple of timeouts in stable, all pass in dev.

**Pull request merged**